python/cudf/cudf/core/index.py

```diff
@@ -1217,6 +1219,130 @@ def memory_usage(self, deep=False):
         """
         return self._values._memory_usage(deep=deep)
 
+    def get_loc(self, key, method=None, tolerance=None):
+        """Get integer location, slice or boolean mask for requested label.
+
+        Parameters
+        ----------
+        key : label
+        method : {None, 'pad'/'fill', 'backfill'/'bfill', 'nearest'}, optional
+            - default: exact matches only.
+            - pad / ffill: find the PREVIOUS index value if no exact match.
+            - backfill / bfill: use NEXT index value if no exact match
+            - nearest: use the NEAREST index value if no exact match. Tied
+                       distances are broken by preferring the larger index
+                       value.
+        tolerance : int or float, optional
+            Maximum distance from index value for inexact matches. The value
+            of the index at the matching location must satisfy the equation
+            abs(index[loc] - key) <= tolerance.
+
+        Returns
+        -------
+        int or slice or boolean mask
+            - If result is unique, return integer index
+            - If index is monotonic, loc is returned as a slice object
+            - Otherwise, a boolean mask is returned
+
```

Comment:
Should we add examples here too? I think it's nice to have examples close by to help with readability. We could probably add the one's pandas uses, and then maybe add a numeric one too.